sched/semaphore/sem_setprotocol.c

```diff
@@ -103,10 +96,11 @@ int nxsem_set_protocol(FAR sem_t *sem, int protocol)
         return -ENOTSUP;
 
       default:
-        break;
+        return -EINVAL;
     }
 
-  return -EINVAL;
+  sem->flags = protocol;
```

Comment:
Why not
```suggestion
  sem->flags &= ~SEM_PRIO_MASK;
  sem->flags = protocol & SEM_PRIO_MASK;
```
?

Reply:
It intends to not mask protocol with SEM_PRIO_MASK, since the upcoming patch will add a new flag:
```
#define SEM_USAGE_MUTEX 0x04
```
and the caller(mutex) can change the usage from sem to mutex explicitly.

Reply:
So mutex will be calling `set_protocol`?

Reply:
Yes, it's called inside nxmutex_init, so it doesn't break the compatibility of user code.